magento/magento2
# Rest API products-render-info return wrong final price for logged in …

### Description (*)
This pull request (PR) will return correct final price of product at product render info Rest API (rest/default/V1/products-render-info?searchCriteria[pageSize]=20&currencyCode=USD&storeId=1) based on customer Token.

### Related Pull Requests
1. Changed "anonymous" to "self" in the Product Render info Rest API
2. Get "CustomerGroupId" from customer token
3. Pass "CustomerGroupId" in productCollection at Rest API function 

### Fixed Issues (if relevant)
1. Fixes  #35757 
### Manual testing scenarios (*)

1. Setup tier price for product A with customer group B, qty 1, price 60$
2. Create new customer account C and assign to customer group B 
3.Open the Postman
4.Add the API URL: {baseurl}/rest/default/V1/products-render-info?searchCriteria[pageSize]=20&currencyCode=USD&storeId=1
5.Method: GET
6.Call API get customer token for customer C
7.Check in response There is a "Final_Price",we get based on customer

<img width="1101" alt="Screenshot 2023-07-04 at 3 14 06 PM" src="https://github.com/magento/magento2/assets/88073031/cad4ca2b-6599-4133-bc8a-4d55a3fd1a55">




### Contribution checklist (*)
